lib/util.js

```diff
@@ -13,28 +12,26 @@ module.exports.exec = function (command, opts, cb) {
 
     opts = opts || {};
 
-    function handleOutput (error, stdout, stderr) {
+    function handleOutput(error, stdout, stderr) {
         // wraps up any errors into a nice package and returns the json value.
-        if (opts.handleError && ((error && error != '') || (stderr && stderr != ''))) {
+        if (opts.handleError && (error || stderr)) {
             cb({
                 stdout: stdout,
                 stderr: [stderr, error].filter(v => !!v).join(''),
                 exitCode: error && error.code,
                 exitSignal: error && error.signal
             });
-        }
-        else {
+        } else {
             // if (stdout) console.log(stdout);
             if (stderr) {
-                if (cb.length == 1) cb(stderr)
+                if (cb.length == 1) cb(stderr);
                 else throw new Error(stderr);
+            } else if (error !== null) {
+                if (cb.length == 1) cb(error, stdout, stderr);
```

Comment:
Good question, I have no idea. This entire file needs to be rewritten IMO. We have so many different custom versions of `exec` floating around. I still need to do a full audit.